user-interface/0001-WALL-wallet.md

```diff
@@ -0,0 +1,179 @@
+
+... so the version of the wallet app I am using works with the network I am using
+
+## Log in to a wallet
+When using a given wallet, I...
+
+- **must** enter wallet name + passphrase only once per "session" <a name="0001-WALL-016" href="#0001-WALL-016">0001-WALL-016</a>
+
+... so that other users of my machine can not use my wallet, and I am not asked to re-enter frequently
+
+## Connecting to Dapps
+When a dapp requests use of a wallet, I...
+
+- **must** be prompted to either select a wallet or dismiss the prompt  <a name="0001-WALL-017" href="#0001-WALL-017">0001-WALL-017</a>
+- `TODO` am required to select what keys of a wallet to grant access too <!--<a name="0001-WALL-018" href="#0001-WALL-018">0001-WALL-018</a>-->
+  - `TODO:` **must** be able to select whole wallet (so that new keys are automatically shared)  <!--<a name="0001-WALL-019" href="#0001-WALL-019">0001-WALL-019</a>-->
+  - `TODO:` **must** select specific [keys](7001-DATA-data_display.md#public-keys) <!-- <a name="0001-WALL-020" href="#0001-WALL-020">0001-WALL-020</a> -->
+  - `TODO` tainted keys **must** be shown as tainted <a name="0001-WALL-021" href="#0001-WALL-021">0001-WALL-021</a>
+- **must** enter wallet passphrase before wallet details are shared <a name="0001-WALL-022" href="#0001-WALL-022">0001-WALL-022</a>
+- **must** be able to retrospectively revoke Dapp's access to Wallet/keys <a name="0001-WALL-023" href="#0001-WALL-023">0001-WALL-023</a>
+
+... so that I can control what public keys are shared with a dapp and what dapps can prompt me to sign transactions 
+
+## Approving transactions
+When a dapp sends a transaction to the wallet for signing and broadcast, I...
+
+- **must** be prompted to approve, reject or ignore the transaction (if auto approve is not on) <a name="0001-WALL-024" href="#0001-WALL-024">0001-WALL-024</a>
+- **must** see the details of the transaction. See [details of transaction](#transaction-detail). <a name="0001-WALL-025" href="#0001-WALL-025">0001-WALL-025</a>
+- **must** see any ignored/dismissed (not rejected or approved) in a transactions area (e.g. history). See [Transactions](#transactions).
+
+... so I can verify that the transaction being sent is the one I want
+
+## View keys balances + positions/accounts
+When looking for a specific balance or asset on a given wallet and network, I...
+
+- **must** see key total balances of an asset for each key <a name="0001-WALL-027" href="#0001-WALL-027">0001-WALL-027</a>
+- **must** see a breakdown of all accounts for an asset on a key <a name="0001-WALL-028" href="#0001-WALL-028">0001-WALL-028</a>
+- **should** see key a summary of balances when switching keys (to help me find the key I am looking for) <a name="0001-WALL-029" href="#0001-WALL-029">0001-WALL-029</a>
+- **should** be able to search by market name, and see all keys with a margin or liquidity account in that market <a name="0001-WALL-030" href="#0001-WALL-030">0001-WALL-030</a>
```

Comment:
market name - do they have to be unique? I'm not sure that markets with the same are rejected. If not then I guess it should show all markets with that name

Reply:
resolved in: https://github.com/vegaprotocol/specs/pull/1181/commits/4665c7e1fe86427ab576ea98028c876b09d53b38